internal/envelope/conn/envelope_conn.go

```diff
@@ -87,15 +88,65 @@ func NewEnvelopeConn(r io.ReadCloser, w io.WriteCloser, h EnvelopeHandler, weave
 // Run interacts with the peer. Messages that are received are
 // handled as an ordered sequence.
 func (e *EnvelopeConn) Run() error {
-	msg := &protos.WeaveletMsg{}
-	for {
-		if err := e.conn.recv(msg); err != nil {
-			return err
+	var group errgroup.Group
+	msgs := make(chan *protos.WeaveletMsg, 100)
+
+	// Spawn a goroutine that repeatedly reads messages from the pipe. A
+	// received message is either an RPC response or an RPC request. conn.recv
+	// handles RPC responses internally but returns all RPC requests. This
+	// goroutine forwards these requests to the goroutine spawned below for
```

Comment:
`The reading goroutine forwards those RPC requests to to goroutine ...`

Reply:
Done.